jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -306,21 +311,48 @@ private void validateMemScope(ResourceScope memScope) throws IllegalStateExcepti
 	 * of the passed-in pointer argument to long.
 	 */
 	private final long memSegmtOfPtrToLongArg(MemorySegment argValue, LinkerOptions options) throws IllegalStateException {
-		UpcallMHMetaData.validateNativeArgRetSegmentOfPtr(argValue, options);
-		addMemArgScope(argValue.scope());
+		try {
+			UpcallMHMetaData.validateNativeArgRetSegmentOfPtr(argValue, options);
+			addMemArgScope(argValue.scope());
```

Comment:
I notice that `addMemArgScope()` could be improved:
```java
	{
		validateMemScope(memArgScope);
		memArgScopeSet.add(memArgScope);
	}
```
(there's no need to check `memArgScopeSet.contains(memArgScope)`).